Tests/Unit/PollerTests.swift

```diff
@@ -0,0 +1,85 @@
+import Quick
+import Nimble
+@testable import RRemoteConfig
+
+class PollerSpec: QuickSpec {
+    override func spec() {
+        describe("init function") {
+            it("sets default delay of 1 hour when no delay param received") {
+                let poller = Poller()
+
+                expect(poller.delay).to(equal(3600.0))
+            }
+
+            it("sets delay to value of param received") {
+                let poller = Poller(delay: 10.0)
+
+                expect(poller.delay).to(equal(10.0))
+            }
+
+            it("does not create a timer") {
+                let poller = Poller()
+
+                expect(poller.timer).to(beNil())
+            }
+        }
+        describe("start function") {
+            it("synchronously executes action closure exactly once") {
+                var actionCalled = 0
+                let poller = Poller()
+
+                poller.start {
+                    actionCalled += 1
+                }
+
+                expect(actionCalled).to(equal(1))
+            }
+
+            it("creates a valid polling timer") {
+                let poller = Poller()
+
+                poller.start {
+                }
+
+                expect(poller.timer?.isValid).to(equal(true))
+            }
+
+            it("creates a timer with expected delay") {
```

Comment:
This test and the one above seem to be testing implementation details. Doesn't the `asynchronously executes the action after the specified delay` test cover this already?

Reply:
Not sure.
The delay and timer properties are currently public (internal scope) so testing their state after function calls seems valid. If I make them private there's no way in Swift of testing their state after init is called.

Reply:
Delay and timer are now private